src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -160,8 +163,10 @@ private CommandResult executeEditStudent(Model model) throws CommandException {
         if (studentToEdit.equals(editedStudent)) {
             throw new CommandException(MESSAGE_UNCHANGED_STUDENT);
         }
-        if (!studentToEdit.isSamePerson(editedStudent) && model.hasStudent(editedStudent)) {
-            throw new CommandException(MESSAGE_DUPLICATE_STUDENT);
+
+        if (model.hasPersonWithSamePhone(editedStudent.getPhone())
```

Comment:
Small nit but maybe you want to put the `!studentToEdit.getPhone().equals(editedStudent.getPhone()` before your `model.hasPersonWithSamePhone(editedStudent.getPhone()` since you did that for your tutor one